pkg/daemon/kata_openshift.go

```diff
@@ -386,13 +417,20 @@ func installRPMs(k *KataOpenShift) error {
 		fmt.Println("Invalid source name")
 		return err
 	}
+
+	defer cleanupHost()
```

Comment:
There's one thing that I'm afraid here and we may need @harche's input.

Consider something fails in alltransports.ParseImageName(), you had an error returned there.
Now you call the cleanupHost() function, which will just go ahead and cleanup the host and return "okay" in case the cleanup is successful.  Wouldn't it override the original error? If it does, is this a problem for you?

Maybe something like this would be more cautious:
```
defer func() {
    cleanupErr = cleanupHost()
    if err == nil {
        return cleanupErr
    }

    return err
```

err, in this case refers to the `err` variable returned from the `installRPMs` function (you'd also have to explicitly set the variable name in the function declaration).

Reply:
Hmm, I think I've tried this and I still got the error from ParseImage(). But now I think there's a different problem. The cleanupHost() can't return an error. Where would it go? I think if an error occurs in cleanupHost() I need to print it right there.

Reply:
Hmm. If it doesn't override the error, we're good.

About "cleanupHost() can't return an error", is it simply because you want to make it non-fatal? Or is there some other reason I'm not seeing?

Reply:
I agree with @fidencio, if you want to have deferred function return err you can use anonymous function like they have shown it here, https://yourbasic.org/golang/defer/#use-func-to-return-a-value